src/MailChimp/Facade.php

```diff
@@ -350,6 +350,43 @@ public function handle_grouping_integrity( $local_groups, $remote_groups, $group
 	}
 
 	/**
+	 * Create all segments necessary
+	 *
+	 * @param array $segments an array of segments
+	 * @return bool
+	 */
+	public function create_segments_from_groups( $segments ) {
+
+		foreach ( $segments as $segment ) {
+
+			try {
+
+				$result = $this->facade->lists->segmentTest( $this->current_list['id'], $segment );
+
+				if ( $result ) {
+
+					// Try adding the segment
+					$args = [
+						'type'         => 'saved',
+						'name'         => md5( $segment['conditions'][0]['value'] ), // 100 byte max so md5 it is
```

Comment:
@mcaskill MailChimp limit the segment name to 100 byte only. What do you think about MD5? It's not pretty in MailChimp back-end but for us it's kind of useful because we will know right away if we need to update the segment if the name doesn't match.... But on the other end I was planning on recreating all segment if we need an update as it would be a little easier to manage. Any input on this?